### ISSUE TITLE
rename `to_json` instance method to `to_hash`

### ISSUE THREAD

**Issue by kplimack** [sendgrid/sendgrid-ruby#334]
#### Issue Summary

All of the classes have a similar method called `to_json` that look like this:
```ruby
def to_json(*)
{
   'enable' => self.enable,
    'enable_text' => self.enable_text
}.delete_if { |_, value| value.to_s.strip == '' }
end
```
But if you look at this, its does not return _JSON_, its returning a compacted hash with each of the class's instance variables enumerated by key/val. 

```ruby
    37: def to_json(*)
    38:   ret = {
    39:     'email' => self.email,
    40:    'name' => self.name
    41:   }.delete_if { |_, value| value.to_s.strip == '' }
 => 43:   binding.pry
    44:   ret
    45: end

[1] pry(#<SendGrid::Email>)> ret
=> {"email"=>"test1@example.com", "name"=>"Example User"}
```

This makes the code somewhat unclear and difficult to understand.
This method and all places that call it should be replaced with the `to_hash` method.

**Comment by haffla** [comment]
Indeed, this was extremely confusing for me, too!! Thanks for tackling this.

**Comment by childish-sambino** [comment]
#362 is in the backlog for review. We're (slowly) working through open PRs. Note that PRs where tests are passing are prioritized first.

**Comment by thinkingserious** [comment]
Since there has been no activity on this issue since March 1, 2020, we are closing this issue. Please feel free to reopen or create a new issue if you still require assistance. Thank you!
